massa-protocol-worker/src/protocol_worker.rs

```diff
@@ -309,33 +318,26 @@ impl ProtocolWorker {
         Ok(self.network_event_receiver)
     }
 
-    /// request the propagation of a set of operations
-    async fn propagate_ops(&mut self, storage: &Storage) {
-        let operation_ids = storage.get_op_refs();
+    /// Announce a set of operations to active nodes who do not know about it yet.
+    /// Side effects:
+    /// - notes nodes as knowing about those operations from now on.
+    /// - empties the buffer of operations to announce.
+    async fn announce_ops(&mut self) {
```

Comment:
* reset the announcement timer
* if the buffer is empty, quit to avoid iterating on nodes

Reply:
> reset the announcement timer

I think I'll just use a tokio interval.

Reply:
@gterzian  the 1st point (resetting the timer) was not addressed in the case where the buffer is full and we send immediately

Reply:
@damip Does it matter? We could also just remove that, or do it first and leave the newly added ops in the buffer for the next tick. We can tweak it later based on how the batching actually turns out I think.

Reply:
* if we only send on timer, sometimes we will get huge batches and extra latency for no reason
* if we send on buffer full but only reset the timer when it ticks, sometimes we will send tiny batches for no reason except that we sent a full buffer just before
* if we reset the timer every time we send, we get the best compromise: low latency when the buffer is full, otherwise maximal buffer size at a given capped latency